xbmc/windowing/Resolution.cpp

```diff
@@ -424,3 +424,50 @@ void CResolutionUtils::PrintWhitelist()
     CLog::Log(LOGDEBUG, "[WHITELIST] whitelisted modes:{}", modeStr);
   }
 }
+
+void CResolutionUtils::GetMaxAllowedResolution(int& width, int& height)
+{
+  if (!CServiceBroker::GetWinSystem()->GetGfxContext().IsFullScreenVideo() &&
+      !CServiceBroker::GetWinSystem()->GetGfxContext().IsFullScreenRoot())
+    return;
+
+  std::vector<RESOLUTION_INFO> resList;
+
+  auto indexList = CServiceBroker::GetSettingsComponent()->GetSettings()->GetList(
+      CSettings::SETTING_VIDEOSCREEN_WHITELIST);
+
+  if (!indexList.empty())
+  {
+    for (const auto& mode : indexList)
+    {
+      RESOLUTION res = CDisplaySettings::GetInstance().GetResFromString(mode.asString());
+      RESOLUTION_INFO resInfo{CServiceBroker::GetWinSystem()->GetGfxContext().GetResInfo(res)};
+      if (resInfo.iWidth > width && resInfo.iHeight > height)
+      {
+        width = resInfo.iWidth;
+        height = resInfo.iHeight;
+      }
+    }
+  }
+  else
+  {
+    std::vector<RESOLUTION> resList;
+    CServiceBroker::GetWinSystem()->GetGfxContext().GetAllowedResolutions(resList);
+
+    for (const auto& res : resList)
+    {
+      RESOLUTION_INFO resInfo{CServiceBroker::GetWinSystem()->GetGfxContext().GetResInfo(res)};
+      if (resInfo.iWidth > width && resInfo.iHeight > height)
+      {
+        width = resInfo.iWidth;
+        height = resInfo.iHeight;
+      }
+    }
+  }
+}
+
+bool CResolutionUtils::IsAdjustRefreshRateEnabled()
```

Comment:
Not sure how it works direct inside add-on. Could it makes sense to allow this call from add-on too with another new function?

Reply:
I think so but i have to check better

Reply:
add GetAdjustRefreshRate